src/hooks/useIsVisible.tsx

```diff
@@ -1,7 +1,10 @@
-import { useRef, useState, useEffect } from 'react';
+import React, { useRef, useState, useEffect } from 'react';
 
-const useIsVisible = (defaultIsVisible = true) => {
-  const sentinelRef = useRef();
+const useIsVisible = (defaultIsVisible = true): [
+  isVisible: boolean,
+  sentinelRef: React.MutableRefObject<HTMLDivElement | null>,
+] => {
+  const sentinelRef = useRef<HTMLDivElement | null>(null);
```

Comment:
[clarification] Will this `sentinelRef` _always_ be applied to a `Div` element? Might it be used with non-`div` elements, too?